### PR TITLE
Unbump core, fileId simulated logic

### PR DESCRIPTION
**Pull request by rpiaggio**
I had to dial back `lucuma-core` since the latest version renames a `DatasetStage` instance, producing incompatibilities.

Also, I make sure the DHS simulator produces filenames that do not conflict (implements @jluhrs 's idea).

### CODE REVIEW

**Comment by jluhrs** [comment]
Just be aware that there will be still a conflict if two fileIds are requested less than 10 seconds apart.

File: modules/server_new/src/main/scala/observe/server/keywords/DhsClientHttp.scala [hunk 1]
```diff
@@ -238,9 +243,15 @@ private class DhsClientSim[F[_]: FlatMap: Logger](date: LocalDate, counter: Ref[
...
 object DhsClientSim {
   def apply[F[_]: Sync: Logger]: F[DhsClient[F]] =
-    (Sync[F].delay(LocalDate.now), Ref.of[F, Int](0)).mapN(new DhsClientSim[F](_, _))
+    Clock[F].monotonic
+      .map(d => Instant.EPOCH.plusNanos(d.toNanos))
+      .map(LocalDateTime.ofInstant(_, ZoneId.systemDefault))
+      .flatMap(apply)
 
-  def apply[F[_]: Sync: Logger](date: LocalDate): F[DhsClient[F]] =
-    Ref.of[F, Int](0).map(new DhsClientSim[F](date, _))
```
Comment (jluhrs) [inline modules/server_new/src/main/scala/observe/server/keywords/DhsClientHttp.scala:244]: This should have worked. Unless DhsClientSim was reinstantiated for every step.

**Review by jluhrs — APPROVED** [review]
LGTM, but we should bump lucuma-core up in all instances, eventually.

**Comment by rpiaggio** [comment]
> LGTM, but we should bump lucuma-core up in all instances, eventually.

Definitely agree.

File: modules/server_new/src/main/scala/observe/server/keywords/DhsClientHttp.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/server_new/src/main/scala/observe/server/keywords/DhsClientHttp.scala:244]: It doesn't work when you restart the server.

**Pull request merged**